paymentsheet/src/main/java/com/stripe/android/paymentelement/confirmation/intent/IntentConfirmationInterceptor.kt

```diff
@@ -280,6 +283,31 @@ internal class DefaultIntentConfirmationInterceptor @Inject constructor(
         )
     }
 
+    private suspend fun waitForIntentCallback(): CreateIntentCallback? {
+        return retrieveCallback() ?: run {
+            val callback = withTimeoutOrNull(INTENT_CALLBACK_FETCH_TIMEOUT.seconds) {
+                var intentCallback: CreateIntentCallback? = null
+
+                while (intentCallback == null) {
+                    delay(INTENT_CALLBACK_FETCH_INTERVAL)
+                    intentCallback = retrieveCallback()
+                }
+
+                intentCallback
+            }
+
+            if (callback != null) {
+                errorReporter.report(ErrorReporter.SuccessEvent.FOUND_CREATE_INTENT_CALLBACK_WHILE_POLLING)
```

Comment:
Will this trigger in the happy path case? I think we only want this event when the first check for the callback was null.